spec/sailfish_iptables_dbus_interface_description.xml

```diff
@@ -0,0 +1,464 @@
+
+<!-- Each function returns a code and descriptive string.
+The returned result codes and string pairs are:
+0	"Ok",
+1	"Invalid IP",
+2	"Invalid port",
+3	"Invalid port range",
+4	"Invalid service name",
+5	"Invalid protocol",
+6	"Invalid policy",
+7	"Invalid file path",
+8	"Cannot process rule",
+9	"Cannot perform operation",
+-->
+
+<!-- 
+Clears given table from iptables. Defaults to filter table.
+Sends signal FirewallCleared
+-->
+<method name="ClearIptablesTable">
+<arg name="table" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Change policy of filter table INPUT chain. Parameter is the same as iptables
+command (ACCEPT/DROP)
+Sends signal PolicyChanged
+-->
+<method name="ChangeInputPolicy">
+<arg name="policy" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Change policy of filter table OUTPUT chain. Parameter is the same as iptables
+command (ACCEPT/DROP)
+Sends signal PolicyChanged
+-->
+<method name="ChangeOutputPolicy">
+<arg name="policy" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Following methods take following parameters as input:
+ NAME		TYPE	DESCRIPTION
+ ip		s		IP address or network with CIDR or dot notation (1.2.3.4/24
+ 					or 1.2.3.4/255.255.255.0)
+ port		q		Port number as unsigned 16bit integer
+ port		s		Port as string, single port or port range, separator is ":"
+ service	s		Service name, case insensitive, 
+ operation	s 		Perform operation add/remove, case insensitive, defaults to
+ 					add if empty string. Add appends the rule to iptables.
+ 					Remove removes the rule from iptables.
+ protocol	s 		tcp/udp, case insensitive (sctp not supported by sailfishos
+ 					iptables yet), usually defaults to TCP as tcp protocol is
+ 					defined first in /etc/services. Can be empty if service/
+ 					port is found from /etc/services.
+
+The methods are named with following convention:
+	Permission|Direction|Parameters
+	
+Permission:
+	Allow			Add rule to iptables with "-j ACCEPT"
+	Deny			Add rule to iptables with "-j DROP"
+	
+Direction:
+	Incoming		Add rule to chain INPUT of filter table
+	Outgoing		Add rule to chain OUTPUT of filter table
+	
+Parameters:
+	Ip			IP address as input
+	With			Combine two parameters
+	Port			Port parameter (integer q)
+	PortRange		Port or port range as parameter (string s)
+	Service			Service name as parameter
+	
+Each method sends signal RuleChanged if the rule was added to iptables.
+-->
+
+<!-- Add IP address or network to filter table INPUT chain as ACCEPT. -->
+<method name="AllowIncomingIp">
+<arg name="ip" type="s" direction="in"/>
+<arg name="operation" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
```

Comment:
This could be (and probably is) the same as `AllowIncomingIpWithPort` with zero port. So it's unnecessary.

Reply:
Doable, maybe 4 different methods (allow in & out, deny in & out) could do the trick. I'll add that to next version with less string inputs.

Reply:
Methods are revamped in 113194ba7b5d66e0b1d2dfaa9fc53558c29b2ebf